packages/in_app_purchase/in_app_purchase/example/lib/main.dart

```diff
@@ -288,14 +287,12 @@ class _MyAppState extends State<_MyApp> {
                     } else {
                       purchaseParam = PurchaseParam(
                         productDetails: productDetails,
-                        applicationUserName: null,
                       );
                     }
 
                     if (productDetails.id == _kConsumableId) {
                       _inAppPurchase.buyConsumable(
-                          purchaseParam: purchaseParam,
-                          autoConsume: _kAutoConsume || Platform.isIOS);
+                          purchaseParam: purchaseParam);
```

Comment:
cc @stuartmorgan 

This lint is weird as well. I'm assuming the original author added the `Platform.isIOS` just in case the `_kAutoConsume` was changed for testing on other platforms. 

This would probably be better as `Platform.isIOS || (!Platform.isIOS && _kAutoConsume)`

Reply:
I looked at the history; it used to be false with a comment saying to change it to true to try auto-consume. It's odd that the comment was dropped when it was switched to true rather than just changed.

Does that formulation actually avoid the warning? I would expect it not to since it trivially reduces to `Platform.isIOS || _kAutoConsume` which is still always true.

I think we should instead change `_kAutoConsume` to be `const bool _kAutoConsume = true || Platform.isIOS;` and add a comment saying:

```
// Auto-consume must be true on iOS. To try without auto-consume on another platform, change `true` to `false` here.
```

Reply:
`const bool _kAutoConsume = true || Platform.isIOS;`
This code conflicts with:
["Const variables must be initialized with a constant value."](https://dart.dev/tools/diagnostic-messages?utm_source=dartdev&utm_medium=redir&utm_id=diagcode&utm_content=const_initialized_with_non_constant_value#const_initialized_with_non_constant_value)

If so, what if we make if final instead of const?
`final bool _kAutoConsume = Platform.isIOS || true;`

Reply:
Yes, changing it to `final` is fine.

Reply:
fixed in [38eaffa](https://github.com/flutter/plugins/pull/6163/commits/38eaffa10e34d442ad98669b697221ea513615a4)